R/rowWilcoxonTests.R

```diff
@@ -72,26 +71,152 @@
 #' cls_perm <- replicate(B, sample(cls))
 #' system.time(fwt <- rowWilcoxonTests(mat, categ = cls_perm, alternative = "two.sided"))
 #' 
-rowWilcoxonTests <- function(mat, categ, 
-                              alternative = c("two.sided", "less", "greater"), 
-                              correct = TRUE) {
+
+rowWilcoxonTests <- function (mat, categ, alternative = c("two.sided", "less", "greater"), 
+                                correct = TRUE) 
+{
+    alternative <- match.arg(alternative)
+    stopifnot(all(categ %in% c(0, 1)))
+    categ <- as.matrix(categ)
+    levels(categ) <- NULL
+    
+    rks <- rowRanks(mat, ties.method = "average")
+    ny <- sum(categ[,1] == 0) 
+    nx <- sum(categ[,1] == 1) 
```

Comment:
Here we implicitly assume that the number of 0 and 1 items is the same for all columns of 'categ'. This is indeed the case if `categ` is generated by permutations of an initial vector. We should raise an error if it's not the case, something like:

```suggestion
    nx <- sum(categ[,1] == 1) 
    nxs <- colSums(categ)
    if (any(nxs != nx)) {
      stop("All columns of 'categ' should have the same number of 0 and 1")
    }
    n <- nrow(categ)
    ny <- n - nx
```

Then we should update the corresponding test to check that an error is indeed raised in this case.

Reply:
Elaborating on the above comment:

The original version of `rowWilcoxonTests` would actually work if the number of 1 in the `categ` matrix was not the same for each columns.  It is not the case anymore, because `nx` and `ny` are scalars and not vectors of length `B`. 

We should make sure that this case is still covered!